sys/include/net/gnrc/nettype.h

```diff
@@ -63,6 +63,9 @@ typedef enum {
 #ifdef MODULE_GNRC_IPV6
     GNRC_NETTYPE_IPV6,          /**< Protocol is IPv6 */
 #endif
+#ifdef MODULE_GNRC_IPV6_EXT
+    GNRC_NETTYPE_IPV6_EXT,      /**< Protocol is IPv6 extension header */
```

Comment:
While I see the need for this `nettype`, I am a bit weary about how this will mess-up the `netreg`. Do we allow for registration to this? And if yes are the demux contexts the same as the next header numbers? Is registration to `GNRC_NETTYPE_IPV6` with the next header number of an extension header still possible?